### ISSUE TITLE
[FR] Work with Algorand to merge in VRF support

### ISSUE THREAD

**Issue by mark-stopka** [jedisct1/libsodium#975]
Algorand [forked libsodium some time ago](https://github.com/algorand/libsodium/tree/draft-irtf-cfrg-vrf-03/src/libsodium/crypto_vrf/ietfdraft03) and implemented [draft-irtf-cfrg-vrf-03](https://tools.ietf.org/html/draft-irtf-cfrg-vrf-03), ( ECVRF-ED25519-SHA512-Elligator2 specifically) VRFs are becoming more commonplace and used nowadays...

I was wondering if they (Algorand) have been in touch with **libsodium** upstream to actually have it merged into mainline or not?

Would this be something you would be willing to merge into your upstream, and if so, what modifications would be required to be made for it to be mergeable?

**Comment by jedisct1** [comment]
The specification is still a draft. Having something that is still a prototype is not great.

Even the hash-to-curve map, which is defined in another document, is still constantly changing.

This particular instantiation can also probably be implemented as a distinct project. All the required primitives are likely to be available in the public API.

VRFs are great, but I have absolutely no use cases for them so far.

That can be revisited, but probably only after the scheme is finalized, instantiations protocols actually use are known, and after the remaining things from the existing roadmap are done.
